content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+---
+title: Best Practices Guide
+section: root
+index: 30
+jump_to:
+  Contents:
+    - Ably 101
+    - Using Ably like a pro
+---
+
+h2(#title). Ably Best Practices Guide
+
+In order to avail the best advantage of our platform, be sure to follow this best practices guide while building your applications using Ably.
+
+h3(#ably-101). Ably 101
+
+Ably Realtime is a cloud-based platform that allows you to easily integrate realtime functionality into your applications. Just plug and play.
+
+h4(#pubsub). Publish and Subscribe
+
+Publish data to a named "channel":/realtime/channels-messages on the Ably Realtime platform, and everyone subscribed to that same channel will receive it.
+
+h4(#notifications). Notify
+
+You can use "Push Notifications":/realtime/push to inform your users about the cool new features you are building in your application.
+
+h4(#presence). Be present
+
+Go online and let others in your application know this by using "Presence":/realtime/presence. You can also see who else is present on a channel even when you yourself are NOT present.
+
+h4(#rest-or-realtime). Remain connected, or not
+
+With the availability of both the "Realtime":/realtime and "REST":/rest libraries, choose what suits your application's needs best and get started.
+
+h4(#auth). Choose the right auth scheme
+
+Use "Basic Authentication":/general/authentication#basic-authentication or "Token Authentication":/general/authentication#token-authentication as well as ensure security by "encrypting your messages.":/realtime/encryption
+
+h3(#using-ably-like-a-pro). Using Ably like a pro
+
+h4. 1. Attaching vs subscribing to a channel
+
+* When your client merely attaches to a channel on Ably, the platform will start sending message data even if your client has not subscribed to the channel. Subscribing to a channel simply registers a listener using which your client can perform additonal operations, however, Ably has not knowledge about this as it only listens for a client to attach to a channel. Bear this in mind while attaching to a channel for publish-only purpose.
+
+h4. 2. Using separate channels when you want messages to be sent to different sets of subscribers.
+
+* Ably pushes all the data published on a channel to every subscriber attached to that channel. Subscribers can choose to only subscribe to events with a certain message name, but that filtering is done on the client side (however, it's useful for things like letting the subscriber trigger different listeners for different events). If you have messages that don't need to be sent to some clients at all, you're often better off using a separate channel for those instead of implementing a filter. This way, you won't be sending unnecessary messages that count towards your package limit.
```

Comment:
Please remove the "(however, it's useful)..." and focus on the problem you are trying to explain to the user.  You have just written a sentence explaining how subsctibe works, focus on that concept, explain it in full, and then perhaps in a separate sentence or paragraph explain where the filtering is useful.